mibidata/tiff.py

```diff
@@ -179,22 +186,17 @@ def write(filename, image, sed=None, optical=None, ranges=None,
             max_value = (341, range_dtype, 1, ranges[i][1])
             page_tags = coordinates + [page_name, min_value, max_value]
 
-            if write_float:
-                target_filename = os.path.join(
-                    filename, '{}.float.tiff'.format(
-                        util.format_for_filename(image.targets[i])))
-            else:
-                target_filename = os.path.join(
-                    filename, '{}.tiff'.format(
-                        util.format_for_filename(image.targets[i])))
+            target_filename = os.path.join(
+                filename, '{}.tiff'.format(
+                    util.format_for_filename(image.targets[i])))
 
             with TiffWriter(target_filename,
                             software=SOFTWARE_VERSION) as infile:
 
-                if write_float:
-                    to_save = image.data[:, :, i].astype(np.float32)
+                if range_dtype == 'I':
+                    to_save = image.data[:, :, i].astype(np.uint16)
                 else:
-                    to_save = image.data[:, :, i]
+                    to_save = image.data[:, :, i].astype(np.float)
```

Comment:
Same thing here. Plus, we don't need to have 2 places to convert the data. Please do a check at the beginning of the file and convert it there.

Reply:
Done.